week-2-wide-perspective/w02e03-pagination-problem/src/pagination-middlewear.js

```diff
@@ -0,0 +1,21 @@
+export const paginate = (model) => {
+  return (req, res, next) => {
+    const { skip, limit } = req.query;
+    let computedSkip = 0;
+    if (skip) {
+      computedSkip = Number(skip);
+    }
+    let computedLimit;
+    if (limit) {
+      computedLimit = Number(limit);
+    }
+    const users = model.find({
```

Comment:
Tutaj się dzieje trochę za dużo. `paginate` nie powinno samo wyciągać danych.
- a co jakbyśmy chcieli poza `.find` uruchomić jakiś `.where` albo inny `filter` ??

Zamiast `model` jako `repository` przygotowałbym tutaj coś na kształt `configa` jakie pola wyciągnąć.
A potem w kontrollerze korzystał z nich tak jak do tej pory.

Np. mógłbyś to wpisać do obiektu: 
```javascript
    paginate: { 
         //...
    }
```

i wtedy dopisać do `req` i korzystać z tego w `controller`

Reply:
Pamiętaj że `middleware` stoi pomiędzy samym requestem a potem `controllerem`, w jakiś sposób przygotowując mu dane. Jednak nie koniecznie samo zapytanie (w tym układzie to po prostu mało elastyczne)